deepomatic/api/http_helper.py

```diff
@@ -157,7 +171,31 @@ def recursive_json_dump(prefix, obj, data_dict, omit_dot=False):
 
         return new_dict
 
-    def make_request(self, func, resource, params=None, data=None, content_type='application/json', files=None, stream=False, *args, **kwargs):
+    def send_request(self, requests_callable, *args, **kwargs):
+        # requests_callable must be a method from the requests module
+        try:
+            # this is the timeout of requests module
+            requests_timeout = kwargs.pop('timeout')
+        except KeyError:
+            requests_timeout = self.requests_timeout
+
+        try:
+            http_retry = kwargs.pop('http_retry')
+        except KeyError:
+            http_retry = self.http_retry
+
+        functor = functools.partial(requests_callable, *args,
+                                    verify=self.verify,
+                                    timeout=requests_timeout, **kwargs)
+
+        if http_retry is not None:
+            return http_retry.retry(functor)
+
+        return functor()
```

Comment:
nit: I would move that in an `else`: it would then be at the same level as with the `http_retry`: much clearer IMO

Reply:
I think it is a matter of taste, I prefer when there is always a non indented return

Reply:
indeed; I was struggling with that and the early return: either both return in if/else, or only one return maybe?
```
if http_retry is not None:
    functor = functools.partial(http_retry.retry, functor)
```